### ISSUE TITLE
Being a collaborator

### ISSUE THREAD

**Issue by jcardi** [gee-community/ee-community#1]
Hi, I'd like to be a collaborator. I see a couple typos that I could fix. Thanks!

**Comment by suredream** [comment]
So nice to have you!

**Comment by suredream** [comment]
BTW, there is a beta version [here](https://github.com/suredream/ee-community), which is already used in my work.

**Comment by gena** [comment]
Jun, you just violated a bunch of licence terms here by copying ee-runner
scripts! This is not how open-source collaborative software development
works.

On 5 May 2017 3:32 p.m., "Jun Xiong" <notifications@github.com> wrote:

BTW, there is a beta version here
<https://github.com/suredream/ee-community>, which is already used in my
work.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/gee-community/ee-community/issues/1#issuecomment-299465295>,
or mute the thread

**Comment by suredream** [comment]
gena:

Would you please tell me what's the correct way to do it? I need some help on this.

I kept your name and license in the project.  I use it in my work and most of the time I use gulp.  I keep ee-runner in it just for run some batch job but I didn't use it often.

I will remove ee-runner from the repo shortly. I now turn to python notebook to do my job.

Thank you for your understanding.

**Comment by suredream** [comment]
gene:

ee-runner code has been removed already.

Please let me know if you have any more concerns.

Thanks,

**Comment by gena** [comment]
If you need to fix a bug of add a feature to the ee-runner code - the best
way to do it is to fork the repo and do pull request or submit an issue.

If you want to create a derived derived work that does not fit the purpose
of the original code - you can better reference it (e.g. as a package), or
at least discuss it first. Blindly copying is error-prone and you also have
to make sure that the license is permissive enough and that all formalities
are met.

If you want to include *any* code example (e.g. any code posted on the
forum by someone) - you need his *explicit* permission to include in the
project under the license terms used for that project. It might be also
better if the author does this himself.

On 5 May 2017 7:00 p.m., "Jun Xiong" <notifications@github.com> wrote:

gene:

ee-runner code has been removed already.

Please let me know if you have any more concerns.

Thanks,

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/gee-community/ee-community/issues/1#issuecomment-299519598>,
or mute the thread

**Comment by suredream** [comment]
Gena:

1. I wrote to you about a PR on ee-runner before.  And I understand you want to keep ee-runner a simple program to run javascript.
2. I didn't include any other's code so far,  most of the code written by myself. I will create a post once it is ready to ask people to contribute themselves.
3. I am still working for a 'community-tools'.  I define it as following components:

- A wiki to guide people to doc/example/resource, no code host, I just added some summmary/comments by myself;
- A framework powered by node-gulp to help people build some big working code from community functions (if we can reach some agreement) and their main program;
- A blog-like website to host documentation and discussion focus on specific topics; forum is good for discussion but not friendly to new users;
- Encourage people to collaborate to the wiki/blog, all these things are open-sourced on github.

I will be on the slack in the next 6 hours if you want to talk.

Right now I don't have too much time for this. I will be a little more flexible at the end of summer.

Cheers,

**Comment by gena** [comment]
> I wrote to you about a PR on ee-runner before. And I understand you want to keep ee-runner a simple program to run javascript.

We've discussed this topic in PR on ee-runner. And my opinion was that it is not a good idea to extend ee-runner to include all code examples, and that it is better to keep these Playground examples in a separate repository (ee-snippets?). But that does not mean that the ee-runner code and examples can be *copied as-is* between repositories with different licenses. Apart from the attribution/licensing question, this is also a bad idea because it will break code history and all bugs will have to be fixed at two places.
 
> I didn't include any other's code so far, most of the code written by myself. I will create a post once it is ready to ask people to contribute themselves.

Just a quick look at examples showed that there are many files not originally written by you:

* https://github.com/gee-community/ee-snippets/blob/master/UI/two-chart-inspector.js - this is Google's code, from UI examples
* https://github.com/gee-community/ee-snippets/blame/9125ea84b5fa5b83695397b2d4f2ca827695f092/UI%20Widgets/Scarbar%20in%20GEE and https://github.com/gee-community/ee-snippets/blame/9125ea84b5fa5b83695397b2d4f2ca827695f092/UI%20Widgets/publisher-ready%20map - this are my examples, I've added my name to the header, but I did not release them yet, neither the license was discussed, I'd probably prefer to clean up everything first before it is distributed
* https://github.com/gee-community/ee-snippets/blame/9125ea84b5fa5b83695397b2d4f2ca827695f092/Misc/Surface%20water%20change%20detection - the code copied from http://github.com/deltares/aqua-monitor
* https://github.com/gee-community/ee-snippets/blame/9125ea84b5fa5b83695397b2d4f2ca827695f092/Datasets/rasterized%20geometries%20on%20PFAF12
* https://github.com/gee-community/ee-snippets/blob/master/Array/array-based-mosaic.js - this is Google's example

... I did not check the rest. But there are files which were originally added without discussion or permission, I would be extremely careful with this. In general - the origin of any source code should be traceable if it was copied, and permission is needed if that is not your code. Maybe one way is to do it via issue tracker (for non-developers). A better way is if the original author can add it. Also, we need some way to review additions, keep things clean. PR on code snipptes repository can be a good way to do this.  

> I am still working for a 'community-tools'. I define it as following components:

What is community-tools? Currently, you have already created a bunch of repositories: gee-community, gee-community-gulp, gae_tools, ee-snippets, ee-community-quickstarts - five repositories and their goals partially overlap.

> A wiki to guide people to doc/example/resource, no code host, I just added some summmary/comments by myself;

+1 for wiki

> A framework powered by node-gulp to help people build some big working code from community functions (if we can reach some agreement) and their main program;
> A blog-like website to host documentation and discussion focus on specific topics; forum is good for discussion but not friendly to new users;

-1 for the blog-like website for, all new users are encouraged to join the forum. 

> Encourage people to collaborate to the wiki/blog, all these things are open-sourced on github.

**Comment by gena** [comment]
Hi Jun,

I did not mean to discourage you, but it is good to avoid some known
(software development in a team) pitfalls to avoid dealing with this later.
It may be better to agree on the list of actions first, who does what,
when, and why. How do we keep everything clean? How is the code reviewed?
Software development in a team can be boring because of many strict rules,
currently we don't have any rules - maybe be ok for the beginning, but not
on a longer term.

Quick review of the repositories (mainly high-level):

   - gee-community - is this a repository where we keep wikipages or also
   snippets? How does this overlap with the ee-community-quickstarts? Why
   should it include examples (deleted today) if there is also ee-snippets
   providing examples?
   - ee-snippets - these code snippets will have to be reviewed, both for
   attribution and for quality.
   - ee-community-quickstarts - how is this different from the
   gee-community wiki?
   - gee-community-gulp - we did not talk about this, but is this a
   repository that replicates the idea I've posted here
   <https://mail.google.com/mail/u/0/#search/gulp++examples/15b34a043d8e0c94>?
   I also noticed a forum post
   <https://mail.google.com/mail/u/0/#search/ee-community-gulp/15ba0e8245d4be4d>
and
   a blog post <https://suredream.github.io/en/2017/04/ee-community-gulp/>
which
   does not mention my original example. No action is neede, but citing the
   origion is a good way to do it. Personally, I don't think this needs to be
   added as a repository right now (see Noel's warning on the forum).
   - gae_tools - what is this? The readme says something vague: "These are
   a bunch of Google Earth Engine Scripts with some tools that may help to
   solve or automatize some processes"


[image: Inline image 1]

This logo is a property of Google. We need permission to use it (if
allowed).

Can we change email to google-earth-engine-developers@googlegroups.com?

---

I'm also very busy the coming months. Let's move on slowly step by step,
avoiding (unintentional) license vialation ;)

Cheers,
Gena

**Comment by suredream** [comment]
Thank you for your reminder,  I did not notice these place which violated the rules. I've fixed these improper use once you finished the merging of the wiki.

It is good to know the correct way to do it. Please let me know what else I can do and whether we can close this issue.

**Comment by gena** [comment]
Thanks!
